File: app/src/main/java/com/study/MyApplication.java
package com.study;

import android.app.Application;


import com.study.util.SharedPreferenceUtils;

public class MyApplication extends Application {

    private static MyApplication instance;
    public MyApplication(){}
    public static MyApplication getInstance(){
        if (instance==null){
            instance=new MyApplication();
        }
        return instance;
    }
    @Override
    public void onCreate() {
        super.onCreate();
        instance=this;
        SharedPreferenceUtils sharedPreferenceUtils=new SharedPreferenceUtils();
        sharedPreferenceUtils.saveLoginInfo("","",false);
    }


}

File: app/src/main/java/com/study/kt/MyApplication.kt
package com.study.kt

import android.app.Application

class MyApplication:Application() {

    /**
     * 静态内部类单例写法
     */
    companion object{
        fun  getInstance()=Helper.instance
    }

    private object Helper{
        val instance=MyApplication()
    }
    override fun onCreate() {
        super.onCreate()

    }
}